pynicotine/transfers.py

```diff
@@ -2085,6 +2113,28 @@ def check_queue_upload_allowed(self, user, addr, filename, real_path, msg):
 
     def check_download_queue(self, *_args):
 
+        if not config.sections["transfers"]["use_download_slots"]:
+            return
+
+        if not self.allow_new_downloads():
+            return
+
+        user = None
+
+        for download in reversed(self.downloads):
+            if download.status != "Locally Queued":
+                continue
+
+            if user is None:
+                user = download.user
+
+            elif user != download.user:
+                continue
```

Comment:
Continuing here will result in the same user being checked again, due to the above condition.